compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -282,6 +288,38 @@ private void visitBuiltinFunctions(BLangPackage pkgNode, BLangFunction function)
         }
     }
 
+    private void visitMockFunctions(BIRPackage birPkg, Map<String, String> mockFunctionMap) {
+        for (BIRFunction function : birPkg.functions) {
+            List<BIRBasicBlock> functionBasicBlocks = function.basicBlocks;
+            for (BIRBasicBlock functionBasicBlock : functionBasicBlocks) {
+                BIRTerminator bbTerminator = functionBasicBlock.terminator;
+                if (bbTerminator.kind.equals(InstructionKind.CALL)) {
+                    mockFunctionMap.forEach((k, v) -> {
+                        String[] mockInfo = k.split(MOCK_ANNOTATION_DELIMITER);
+                        if (mockInfo.length != 2) {
+                            return;
+                        }
+                        if (((BIRTerminator.Call) bbTerminator).name.getValue().equals(mockInfo[1])) {
+                            ((BIRTerminator.Call) bbTerminator).name = getMockFunctionName(v, birPkg);
+                            if (!mockInfo[0].equals(".")) {
+                                ((BIRTerminator.Call) bbTerminator).calleePkg = function.pos.src.pkgID;
```

Comment:
Since your are doing the same casting operation `((BIRTerminator.Call) bbTerminator)` multiple times,  better to do it once and keep it a local variable.

Reply:
Implemented a local variable for this. Will continue to refactor this nested loop case.